recipes/glog/all/conanfile.py

```diff
@@ -18,88 +21,94 @@ class GlogConan(ConanFile):
         "fPIC": [True, False],
         "with_gflags": [True, False],
         "with_threads": [True, False],
+        "with_unwind": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
         "with_gflags": True,
         "with_threads": True,
+        "with_unwind": True,
     }
 
-    exports_sources = ["CMakeLists.txt", "patches/**"]
-    generators = "cmake", "cmake_find_package"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def export_sources(self):
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if self.settings.os not in ["Linux", "FreeBSD"] or Version(self.version) < "0.5.0":
+            del self.options.with_unwind
 
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
         if self.options.with_gflags:
             self.options["gflags"].shared = self.options.shared
 
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
     def requirements(self):
         if self.options.with_gflags:
             self.requires("gflags/2.2.2")
+        # 0.4.0 requires libunwind unconditionally
+        if self.options.get_safe("with_unwind") or (Version(self.version) < "0.5.0" and self.settings.os in ["Linux", "FreeBSD"]):
+            self.requires("libunwind/1.6.2")
 
     def build_requirements(self):
-        if tools.Version(self.version) >= "0.6.0":
+        if Version(self.version) >= "0.6.0":
             self.build_requires("cmake/3.22.3")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["WITH_GFLAGS"] = self.options.with_gflags
+        tc.variables["WITH_THREADS"] = self.options.with_threads
+        tc.variables["BUILD_SHARED_LIBS"] = self.options.shared
+        if Version(self.version) >= "0.5.0":
+            tc.variables["WITH_PKGCONFIG"] = True
+            if self.settings.os == "Emscripten":
+                tc.variables["WITH_SYMBOLIZE"] = False
+                tc.variables["HAVE_SYSCALL_H"] = False
+                tc.variables["HAVE_SYS_SYSCALL_H"] = False
+            else:
+                tc.variables["WITH_SYMBOLIZE"] = True
+            tc.variables["WITH_UNWIND"] = self.options.get_safe("with_unwind", default=False)
+        tc.variables["BUILD_TESTING"] = False
+        tc.variables["WITH_GTEST"] = False
+        tc.generate()
```

Comment:
```suggestion
        tc.generate()
        tc = CMakeDeps(self)
        tc.generate()
```
It will generate all dependencies cmake files, based on their conan recipes. More info: https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmakedeps.html

Reply:
I had to revert that, it broke the build.